tests/cost_tests.cpp

```diff
@@ -27,4 +28,76 @@ TEST_CASE("Linear Cost Test", "[cost]") {
     torc::LinearCost cost2 = torc::LinearCost<float>(q2, name2);
     REQUIRE(cost2.Evaluate(x3) == 26);
     REQUIRE(cost2.Gradient() == q2);
+}
+
+
+TEST_CASE("Finite Difference Cost Test") {
+    SECTION("Basic grad and hess test") {
+        std::function<double(Eigen::VectorXd)> fn = [](const Eigen::VectorXd& x) {
+            return x.squaredNorm();
+        };;
+        torc::FiniteDiffCost cost1 = torc::FiniteDiffCost<double>(fn, 2);
+
+        Eigen::Vector2d in = {1, 2};
+        REQUIRE(cost1.Evaluate(in) == 5);
+
+        Eigen::Vector2d grad_expected = {2, 4};
+        REQUIRE(cost1.Gradient(in).isApprox(grad_expected, 1e-8));
```

Comment:
FYI Catch2 has support for comparing floats and approximate equality. See https://github.com/catchorg/Catch2/blob/devel/docs/comparing-floating-point-numbers.md

Reply:
Thanks for the heads-up!